synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -150,6 +152,10 @@ export class SynopsysToolsParameter {
       }
     }
 
+    // Set Coverity or Blackduck Arbitrary Arguments
+    polData.data.coverity = this.setCoverityArbitraryArgs();
+    polData.data.blackduck = this.setBlackDuckArbitraryArgs();
```

Comment:
Can we set these values for Coverity and BlackDuck conditionally based on Polaris assessment type? Like - 
`if (assessmentType === 'SCA') {
    polData.data.blackduck = this.setBlackDuckArbitraryArgs();
} else if (assessmentType === 'SAST') {
    polData.data.coverity = this.setCoverityArbitraryArgs();
}`

With this we can only pass the params that are required for specific assessment type. Otherwise if the user wants to run the SCA scan only but if SAST values are also set in YAML/Classic Editor that values may also get passed in input.json

Reply:
There should not be any validation based on Assessment types. 
cc: @sig-tithi

Reply:
If I am not wrong there should be check for assessment types like 

1. **Blackduck** arbitrary params will only pass if assessment type is **SCA**
2. **Coverity** arbitrary params will only pass if assessment type is **SAST**
3. Both **Blackduck** & **Coverity** arbitrary params will pass if assessment type is **SCA,SAST**

Reply:
What is the current behaviour if user selects Polaris SCA and sends coverity args- is it getting ignored b by bridge or thrown any error?

Reply:
it is ignored by bridge